service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/RequestStatementExtractor.java

```diff
@@ -128,4 +138,196 @@ public static String extractNoteTime(Annotation annotation) {
         }
         return StringUtils.EMPTY;
     }
+
+    public static String extractSupportingInfo(MessageContext messageContext, Reference reference) {
+        IIdType referenceId = reference.getReferenceElement();
+        String resourceType = referenceId.getResourceType();
+
+        switch (resourceType) {
+            case "DocumentReference":
+                return extractDocumentReference(messageContext, reference);
+            case "Observation":
+                return extractObservation(messageContext, reference);
+            case "MedicationRequest":
+                return extractMedicationRequest(messageContext, reference);
+            case "ReferralRequest":
+                return extractReferralRequest(messageContext, reference);
+            case "DiagnosticReport":
+                return extractDiagnosticReport(messageContext, reference);
+            default:
+                throw new EhrMapperException("SupportingInfo Reference not of expected Resource Type");
```

Comment:
Don't throw an exception here, if the resource type is unsupported we just ignore it.